e2e/utils/index.ts

```diff
@@ -468,7 +468,7 @@ export function readJson(f: string): any {
   try {
     return JSON.parse(content);
   } catch {
-    return JSON.parse(stripJsonComments(content));
+    return parseJson(content);
```

Comment:
Did you perhaps compare performance to `strip-json-comments'?

We also use this logic (try...catch) for linter since most of our JSON files are not expected to have any comments, and performance gain was noticeable.

I would extract this logic to utilities and replace all the instances of `stripJsonComments` and `JSON.parse` (where applicable) with wrapper function.

Reply:
@meeroslav I compared the performance and it is very similar but `jsonc-parser` it slightly faster. I tested with the workspace.json file of the Nx repo and a version in which i added about 30 comments to it.  

Results:
```
CPU: Intel(R) Core(TM) i7-9750H CPU
RAM: 16GB

Parse workspace.json with comments:
---------------------------------------
JSON.parse() and stripJsonComments()
20.1066ms
---------------------------------------
jsonc-parser parse()
19.2196ms
---------------------------------------

Only stripComments from workspace.json with Comments:
---------------------------------------
stripJsonComments()
15.7106ms
---------------------------------------
jsonc-parser stripComments()
11.2609ms
---------------------------------------

Parse workspace.json without comments:
---------------------------------------
JSON.parse()
2.0392
---------------------------------------
jsonc-parser parse
8.245
---------------------------------------
```

We can add the try catch block with the JSON.parse to the `parseJson` method to use the faster JSON.parse when possible. but this costs us extra ~2ms if we want to parse a JSON file with comments.

Reply:
Thanks, that's in-line with my investigations for `stripJsonComments`. The way I see it, we have two scenarios:

**Optimistic**
- no downside on pure json
- up to 10% (worst case, when comment is found last) overwork for JSONs with comments

**Pessimistic**
- no downside on JSON with comments
- 900% overwork for pure JSONs 

Files Nx generates are without comments and most of the JSONs we will encounter are pure. Having pessimistic approach is a heavy price to pay for the small percentage of commented JSONs we deal with.

I would definitely go with single `parseJson` function in dev-kit that would be used everywhere.
```typescript
function parseJson<T extends object = any>(content: string): T {
  try {
    return JSON.parse(content);
  } catch {
    const errors: ParseError[] = [];
    const result: T = jsoncParser.parse(content, errors);
    if (errors.length > 0) {
      const { error, offset } = errors[0];
      throw new Error(
        `${printParseErrorCode(error)} in JSON at position ${offset}`
      );
    }
    return result;
  }
}
```

Optionally, if we know the JSON file to have comments (I don't see such scenario, but could happen) we can provide flag that would jump straight to using jsonc-parser.
```typescript
parseJson(content: string, expectComments?: boolean)
```

@vsavkin, @FrozenPandaz what do you guys think?

Reply:
@meeroslav I totally aggree. What do you think of the following implementation. It allow us to disallow comments in files but also expect comments. It also allows us to add more options if needed without changing the API. 

```ts
export interface JsonParseOptions {
  /**
   * Expect JSON with javascript-style
   */
  expectComments?: boolean;
  /**
   * Disallow javascript-style
   */
  disallowComments?: boolean;
}

/**
 * Parses the given JSON string and returns the object the JSON content represents.
 * By default javascript-style comments are allowed.
 *
 * @param input JSON content as string
 * @param options JSON parse options
 * @returns Object the JSON content represents
 */
export function parseJson<T extends object = any>(
  input: string,
  options?: JsonParseOptions
): T {
  try {
    if (options?.expectComments !== true) {
      return JSON.parse(input);
    }
  } catch (error) {
    if (options?.disallowComments === true) {
      throw error;
    }
  }

  const errors: ParseError[] = [];
  const result: T = parse(input, errors, {
    disallowComments: options?.disallowComments ?? false,
  });

  if (errors.length > 0) {
    const { error, offset } = errors[0];
    throw new Error(
      `${printParseErrorCode(error)} in JSON at position ${offset}`
    );
  }

  return result;
}
```

I also would recommend to define the methods in `@nrwl/tao` and just reexport them from `@nrwl/devkit`. If you look at the dependency graph `tao` is the root project which all other packages depend on. Through that the JSON parse and serialization behaviour can be normalized in all packages. 

>I would extract this logic to utilities and replace all the instances of stripJsonComments and JSON.parse (where applicable) with wrapper function.

I totally agree and that was actually the plan of this PR. As i mentioned in the PR description if this PR gets merged i plan to open some follow up PRs which then aim to replace JSON.parse and JSON.stringify with the wrapper functions.

Reply:
I like it! Thanks!